pkg/wallet/wallet_create_action_test.go

```diff
@@ -968,6 +970,59 @@ func (s *WalletTestSuite) TestWalletCreateAction_SendWithAsRetryOfProcessAction(
 }
 
 func (s *WalletTestSuite) TestWalletCreateActionByBobBasedOnAliceCreateAction() {
+	s.Run("alice and bob uses the different storage", func() {
+		t := s.T()
+		const topUpValue = testValueForFunding
+
+		// given:
+		given, cleanup := testabilities.Given(t)
+		defer cleanup()
+
+		// and:
+		aliceWallet := given.Wallet().WithOwnStorage().ForUser(testusers.Alice)
+
+		// and:
+		bobWallet := given.BobWalletWithStorage(s.StorageType)
+
+		// and:
+		_, _ = given.Faucet(aliceWallet).TopUp(topUpValue)
+
+		// when:
+		aliceArgs := fixtures.DefaultWalletCreateActionArgs(t, walletargs.WithLockingScript(script.Script{
+			script.Op3,
+			script.OpEQUAL,
+		}))
+
+		firstResult, err := aliceWallet.CreateAction(t.Context(), aliceArgs, fixtures.DefaultOriginator)
+
+		// then:
+		require.NoError(t, err)
+		require.NotEmpty(t, firstResult.Tx, "Alice wallet should return transaction BEEF bytes")
+		require.NotEmpty(t, firstResult.Txid, "Alice wallet should return transaction ID")
+
+		// when:
+		bobsArgs := fixtures.DefaultWalletCreateActionArgs(t,
+			walletargs.WithInputs([]sdk.CreateActionInput{
+				{
+					Outpoint:         transaction.Outpoint{Txid: firstResult.Txid, Index: 0},
+					InputDescription: "got from alice",
+					UnlockingScript:  script.Script{script.Op3},
+				},
+			}),
+			walletargs.WithNoOutputs(),
+			walletargs.WithInputBEEF(firstResult.Tx),
+		)
+
+		bobsResult, err := bobWallet.CreateAction(t.Context(), bobsArgs, fixtures.DefaultOriginator)
+
+		// then:
+		require.NoError(t, err)
+		internaltestabilities.AssertAtomicBEEFState(t, bobsResult.Tx[:], internaltestabilities.ExpectedBeefTransactionState{
+			ID:         firstResult.Txid.String(),
+			DataFormat: to.Ptr(transaction.RawTx),
+		})
```

Comment:
It would be good to check the state of Bob's storage:

```suggestion
		})
		// and check db state:
		thenState := testabilities.ThenWalletState(t, bobWallet)
		// ....
```

And additionally, but optionally - to try to make bobWallet `createAction` to ensure Bob really is able to spend newly received funds.

Reply:
@mgosek-4chain those additional checks are important part, please add them